server/controllers/calls.js

```diff
@@ -66,42 +108,28 @@ function afterPutCallAttempt(res, outcome, contact_id, attempt_num, campaign_id)
   }
   return res.status(200).json({ message: 'call log successfully updated' });
 }
-
-export function assignCall(req, res) {
-  const user_id = parseInt(req.params.id, 10);
-  const user_campaign_id = parseInt(req.params.campaign_id, 10);
-
-  userHasJoinedCampaign(user_id, user_campaign_id)
-  .then((userHasJoined) => {
-    if (userHasJoined) {
-      return callsService.assignCall({
-        campaign_id: user_campaign_id,
-        user_id
-      }).then((call) => {
-        if (call) {
-          return res.status(200).json(call);
-        }
-
-        return res.status(404).json({ message: 'no calls available' });
-      }).catch(err => console.log('Could not assign call:', err));
-    }
-    return res.status(401).json({ message: 'User has not joined that campaign' });
-  }).catch(err => console.log(err));
-}
+/* ======== END HELPERS ========== */
 
 export function recordAttempt(req, res) {
-  const { outcome, notes, responses } = req.body;
+  const { outcome, notes, responses, status: newStatus } = req.body;
   let parsedResponses;
-  try {
-    parsedResponses = JSON.parse(responses);
-  } catch (err) {
-    return res.status(400).json({ message: 'Invalid JSON object' });
+  // responses will not exist in a status update for HUNG_UP and IN_PROGRESS
+  if (newStatus === 'ATTEMPTED') {
+    if (!responses || !outcome) {
+      res.staus(404).json({ message: 'update request with a status of ATTEMPTED must have response object and outcome string' });
```

Comment:
400 is a better code. 404 is reserved for when things aren't found/don't exist.

Reply:
Got it ✔️